docs/source/configuration.rst

```diff
@@ -88,6 +109,79 @@ in projects which utilise a lot of complicated templating.
 
 .. _templateconfig:
 
+Rule Configuration
+------------------
+
+Rules can be configured with the :code:`.sqlfluff` config files.
+
+Common rule configurations can be set in the :code:`[sqlfluff:rules]` section.
+
+For example:
+
+.. code-block:: cfg
+
+   [sqlfluff:rules]
+   tab_space_size = 4
+   max_line_length = 80
+   indent_unit = space
+   comma_style = trailing
+   allow_scalar = True
+   single_table_references = consistent
+   unquoted_identifiers_policy = all
+
+Rule specific configurations are set in rule specific subsections.
+
+For example, enforce that keywords are upper case by configuring the rule
+:class:`L010 <sqlfluff.core.rules.Rule_L010>`:
+
+.. code-block:: cfg
+
+    [sqlfluff:rules:L010]  # Keywords
+    capitalisation_policy = upper
+
+All possible options for rule sections are documented in :ref:`ruleref`.
+
+For an overview of the most common rule configurations that you may want to
+tweak, see `Default Configuration`_ (and use :ref:`ruleref` to find the
+available alternatives).
+
+Enabling and Disabling Rules
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To disable individual rules, set :code:`exclude_rules` in the top level section
+of sqlfluff configuration. The value is a comma separated list of rule ids.
+
+For example, to disable the rules :class:`L010 <sqlfluff.core.rules.Rule_L022>`
+and :class:`L010 <sqlfluff.core.rules.Rule_L027>`:
+
+.. code-block:: cfg
+
+    [sqlfluff]
+    exclude_rules = L022, L027
+
+To enable individual rules, configure :code:`rules`, respectively.
+
+For example, to enable :class:`L010 <sqlfluff.core.rules.Rule_L027>`:
+
+.. code-block:: cfg
+
+    [sqlfluff]
+    rules = L027
+
+^ **TODO** does this make sense only if overriding an exclusion from a higher level
+configuration? Or what does this conf mean? How is it interpreted if
+:code:`exclude_rules = None` but :code:`exclude_rules != None`?
```

Comment:
Do you know how this is? If not, I could try to look it up but not feeling super confident :)

Reply:
Someone looked this up a few days ago and exclude takes priority over rules if that’s what you’re asking.

I think it only makes sense to either to explicitly include some rules, or to include all and explicitly include some of them. Mixing the two doesn’t make sense IMHO.

Reply:
> exclude takes priority over rules

Oh. So then `rules` only makes sense to be set if `exclude_rules = None`? If `exclude_rules` is set to any value, `rules` will be entirely ignored?

Reply:
Is there a concept like this though: "overriding an exclusion from a higher level
configuration"? And is it possible to have `.sqlfluff` files in folder hierarchy that overwrite some particular configurations? If yes, then I could imagine `rules` to be set on highest level, but then one could exclude some of those rules with `exclude_rules` in the scope of some subdirectory. Or am I just imagining something that is not even a feature? :)

Reply:
Ah that’s true. Yes you could do that.